src/components/Auth/CP/CPAuthProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import CPAuth from 'lib/CPAuth/CPAuth';
+import { OAuth2Events } from 'lib/OAuth2/OAuth2';
+import { IOAuth2User } from 'lib/OAuth2/OAuth2User';
+import React, { ReactElement, useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  loadUserSuccess,
+  userExpired,
+  userExpiring,
+  userSignedOut,
+} from 'stores/cpauth/actions';
+
+interface ICPAuthProviderProps extends React.PropsWithChildren<{}> {}
+
+const CPAuthProvider: React.FC<ICPAuthProviderProps> = ({ children }) => {
```

Comment:
This component sits as a direct child of the `Layout` component, and it acts as a mapping between the Control Plane authentication provider and the Redux store.